lib/IntermediateRepresentation/procedure.h

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) 2019, NVIDIA CORPORATION.  All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef FORTRAN_INTERMEDIATEREPRESENTATION_PROCEDURE_H_
+#define FORTRAN_INTERMEDIATEREPRESENTATION_PROCEDURE_H_
+
+#include "mixin.h"
+#include "program.h"
+#include "region.h"
+#include "llvm/ADT/StringRef.h"
+#include "llvm/ADT/Twine.h"
+
+namespace Fortran::IntermediateRepresentation {
+
+struct Program;
+struct Region;
+struct GraphWriter;
+
+struct Procedure final : public llvm::ilist_node<Procedure>,
+                         public ChildMixin<Procedure, Program> {
+  friend GraphWriter;
+  friend Program;
+  friend Region;
+  using BasicBlockListType = llvm::iplist<BasicBlock>;
+  using RegionListType = llvm::iplist<Region>;
+  using iterator = BasicBlockListType::iterator;
+  using const_iterator = BasicBlockListType::const_iterator;
+  using reverse_iterator = BasicBlockListType::reverse_iterator;
+  using const_reverse_iterator = BasicBlockListType::const_reverse_iterator;
+
+  Procedure(const Procedure &) = delete;
+  Procedure &operator=(const Procedure &) = delete;
+  ~Procedure();
+  BasicBlockListType &GetBlocks() { return basicBlockList_; }
+  BasicBlockListType &getSublist(BasicBlock *) { return GetBlocks(); }
```

Comment:
What does `getSublist()` do?

If the argument pointer cannot be null, it might be more clear if it were a reference.